code/game/objects/items/holy_weapons.dm

```diff
@@ -263,30 +262,89 @@
 	return (BRUTELOSS|FIRELOSS)
 
 /obj/item/nullrod/attack_self(mob/user)
-	if(user.mind && (user.mind.holy_role) && !reskinned)
+	if(user.mind && (user.mind.holy_role) && !current_skin)
 		reskin_holy_weapon(user)
 
 /obj/item/nullrod/proc/reskin_holy_weapon(mob/M)
-	var/obj/item/nullrod/holy_weapon
-	var/list/holy_weapons_list = typesof(/obj/item/nullrod)
-	var/list/display_names = list()
-	for(var/V in holy_weapons_list)
-		var/obj/item/nullrod/rodtype = V
-		if (initial(rodtype.chaplain_spawnable))
-			display_names[initial(rodtype.name)] = rodtype
-
-	var/choice = input(M,"What theme would you like for your holy weapon?","Holy Weapon Theme") as null|anything in sortList(display_names, /proc/cmp_typepaths_asc)
-	if(QDELETED(src) || !choice || M.stat || !in_range(M, src) || M.incapacitated() || reskinned)
-		return
-
-	var/A = display_names[choice] // This needs to be on a separate var as list member access is not allowed for new
-	holy_weapon = new A
-
-	SSblackbox.record_feedback("tally", "chaplain_weapon", 1, "[choice]")
-
-	if(holy_weapon)
-		holy_weapon.reskinned = TRUE
+	unique_reskin = list(
+		"Null Rod" = /obj/item/nullrod,
+		"God Hand" = /obj/item/nullrod/godhand,
+		"Red Holy Staff" = /obj/item/nullrod/staff,
+		"Blue Holy Staff" = /obj/item/nullrod/staff/blue,
+		"Claymore" = /obj/item/nullrod/claymore,
+		"Dark Blade" = /obj/item/nullrod/claymore/darkblade,
+		"Sacred Chainsaw Sword" = /obj/item/nullrod/claymore/chainsaw_sword,
+		"Force Weapon" = /obj/item/nullrod/claymore/glowing,
+		"Hanzo Steel" = /obj/item/nullrod/claymore/katana,
+		"Extradimensional Blade" = /obj/item/nullrod/claymore/multiverse,
+		"Light Energy Sword" = /obj/item/nullrod/claymore/saber,
+		"Dark Energy Sword" = /obj/item/nullrod/claymore/saber/red,
+		"Nautical Energy Sword" = /obj/item/nullrod/claymore/saber/pirate,
+		"UNREAL SORD" = /obj/item/nullrod/sord,
+		"Reaper Scythe" = /obj/item/nullrod/scythe,
+		"High Frequency Blade" = /obj/item/nullrod/scythe/vibro,
+		"Dormant Spellblade" = /obj/item/nullrod/scythe/spellblade,
+		"Possessed Blade" = /obj/item/nullrod/scythe/talking,
+		"Possessed Chainsaw Sword" = /obj/item/nullrod/scythe/talking/chainsword,
+		"Relic War Hammer" = /obj/item/nullrod/hammmer,
+		"Chainsaw Hand" = /obj/item/nullrod/chainsaw,
+		"Clown Dagger" = /obj/item/nullrod/clown,
+		"Pride-struck Hammer" = /obj/item/nullrod/pride_hammer,
+		"Holy Whip" = /obj/item/nullrod/whip,
+		"Atheist's Fedora" = /obj/item/nullrod/fedora,
+		"Dark Blessing" = /obj/item/nullrod/armblade,
+		"Unholy Blessing" = /obj/item/nullrod/armblade/tentacle,
+		"Carp-Sie Plushie" = /obj/item/nullrod/carp,
+		"Monk's Staff" = /obj/item/nullrod/claymore/bostaff,
+		"Arrythmic Knife" = /obj/item/nullrod/tribal_knife,
+		"Unholy Pitchfork" = /obj/item/nullrod/pitchfork,
+		"Egyptian Staff" = /obj/item/nullrod/egyptian,
+		"Hypertool" = /obj/item/nullrod/hypertool,
```

Comment:
Do these have to be manually typed out? You can't generate this using subtypesof or something?

Reply:
subtypesof just returns the typepath, won't be useful for me since even if I use that, the 'names' of the items in this list are different from their names in their code.